docs/developer/architecture/centralized-cli.rst

```diff
@@ -0,0 +1,145 @@
+- CLI provides an exhaustive help, per action, with relevant examples
+- CLI is not interactive (except for static user provisioning)
+- CLI should not require password input
+- CLI provides a dryrun mode for intrusive operations
+- CLI provides a verbose (or debug) mode
+- CLI implementation relies on secure APIs
+- CLI support <action> completion for easy discovery
+- CLI output is standardized and human readable by default
+- CLI output can be formatted in JSON or YAML
+- CLI can be executed from outside of the cluster (except maybe for very first
+  install script)
+
+When it is possible, it would make sense to leverage kubectl plugin
+
+Requirements
+------------
+
+Cluster Resources Administration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+tools: kubectl
+
++------------+------------------+-----------------------------------------+
+| Resource   | action           | parameters                              |
++============+==================+=========================================+
+| node       | add              | name, ssh-user, hostname or ip,         |
+|            |                  | ssh port, ssh-key-path, sudo-required,  |
+|            |                  | roles, manifest-template-path           |
++------------+------------------+-----------------------------------------+
+| the manifest-template-path is the path to the manifest describing the   |
+| resource. Everything that is in the command line, override the manifest |
+| content. This apply to Node and Volumes.                                |
++------------+------------------+-----------------------------------------+
+| node       | edit             | name, hostname (to change node hostname)|
++------------+------------------+-----------------------------------------+
+| node       | deploy(or apply?)| <list of nodes>, dry-run                |
++------------+------------------+-----------------------------------------+
+| node       | delete, drain,   | <list of nodes>                         |
+|            | taint, label     |                                         |
++------------+------------------+-----------------------------------------+
+| volume     | add              | name, nodeName, storageClassName,       |
+|            |                  | devicePath, manifest-template-path,     |
+|            |                  | labels                                  |
++------------+------------------+-----------------------------------------+
+| volume     | delete, label    | name                                    |
++------------+------------------+-----------------------------------------+
+
+
+Cluster Administration
+^^^^^^^^^^^^^^^^^^^^^^
+
+tools: mk8sctl
+
++------------+------------+-----------------------------------------------+
+| Resource   | action     | parameters                                    |
++============+============+===============================================+
+| cluster    | install    |                                               |
```

Comment:
What `install cluster` mean ? It's bootstrap right ? Or it's something with a platform description file and this do bootstrap + expansions ?

Also I think we want a command to `install bootstrap`.

Since we use `<action> <resource>` style I think you should have `deploy` or `install` but not a mix because command will look like 
`kubectl install cluster`

Reply:
you are right, it should be `install` `bootstrap`.
Not sure I understand your second comment

Reply:
For node you have `node` `deploy` and for bootstrap it's `bootstrap` `install` for me you should have only `deploy` or only `install`

Reply:
okay understood

Reply:
done